### DISCUSSION TITLE
How to use JSON string in REST API bindings

### DISCUSSION THREAD

**Discussion by jlangree** [Budibase/budibase#8496] [Help] [answered]
It appears I cannot use a JSON string as a binding for a REST API call. Is there some way around this?

It seems to work if the backslashes are tripled (i.e. `"{ \\\"stuff\\\": \\\"things\\\" }"` ), but there doesn't seem to be a way to return that string from a JavaScript binding function. No matter how many backslashes are added in, it always comes out with normal, single-backslashed JSON string. There must be special handling of escape characters somewhere?

![image](https://user-images.githubusercontent.com/48861088/199555241-51c503ec-6d74-44fe-90a5-63f2064f69eb.png)
![image](https://user-images.githubusercontent.com/48861088/199555509-6f455bc4-2f1c-4e33-b7b4-8cd09209df2a.png)
![image](https://user-images.githubusercontent.com/48861088/199555571-4f83116c-977f-4a18-a857-c045cc9f7561.png)

**Comment by jlangree** [comment, marked as answer]
While I still haven't found a way to successfully send a JSON string as a binding in a request, I did figure out how to send the full object in a post body. You simply omit the quotes in the post body. I hate how long that took me to figure out. I _just_ upgraded to v2.0- maybe this is a new-ish feature?

JS function to send form data to binding for REST API:
```js
const addStudent = $("Role Form.Fields.addStudentRole")
const addOwner = $("Role Form.Fields.addOwnerRole")
const addCustom = $("Role Form.Fields.addCustomRole")

const studentRole = addStudent ? {
	name: 'student',
	description: 'for students in the group',
	includeInReports: true
} : null

const ownerRole = addOwner ? {
	name: 'owner',
	description: 'group owner',
	includeInReports: false
} : null

const customRole = addCustom ? {
	name: $("Role Form.Fields.roleName"),
	description: $("Role Form.Fields.roleDescription"),
	includeInReports: $("Role Form.Fields.roleIncludeInReports")
} : null

const roles = [studentRole, ownerRole, customRole].filter(r => r !== null).map(r => ({
	...r,
	includedInGroups: [{ id: $("URL.id") }],
	createdAt: new Date().toISOString(),
	default: false
}))

return { input: roles }
```

REST API post body:
```json
{
  "variables": {{ yourJsonObjBinding }}
}
```

**Comment by ChuTungLinh** [comment]
so i've found a way to successfully send  JSON string 

json parse:
```javascript
let a = JSON.stringify($("State.AnswerSet"))
let result = a.replace(/"/g, '\\\"') 
return result
``` 
it is because for some reason budibase require 3 "\\\" to accept it dont ask me y cause idk either i just stumble upon it

API body:
```javascript
{
"variables": "{{ yourJsonObjBinding }}"
}
```
